doc/src/labelmap.rst

```diff
@@ -0,0 +1,111 @@
+.. index:: labelmap
+
+labelmap command
+==================
+
+Syntax
+""""""
+
+.. code-block:: LAMMPS
+
+   labelmap interaction numeric-type type-label ... keyword values
+
+* interaction = *atom* or *bond* or *angle* or *dihedral* or *improper*
+* one or more numeric-type/type-label pairs may be appended
+* keyword = *map/assign*
+
+  .. parsed-literal::
+
+       *map/assign* value = mapID
+         mapID = name of auxiliary label map
+
+Examples
+""""""""
+
+.. code-block:: LAMMPS
+
+   labelmap atom 3 carbon
+   labelmap bond 1 [c1][c2] 2 [c1][hc]
+
+Description
+"""""""""""
+
+Define or replace the alphanumeric type label associated with a
+numeric atom, bond, angle, dihedral or improper type. Type labels are
+strings that can be assigned to each interaction type. Pairing each
+numeric type used by LAMMPS with a character-based type can be
+helpful in a variety of situations. For example, type labels can make
+various inputs more readable and compatible with other inputs (data
+files, molecule templates, etc.), particularly if your model's force
+field uses alphanumeric atom types. The default type label map can
+also be defined by :doc:`data files <read_data>` using the relevant
+Type Label sections.
+
+.. note::
+
+   If substituting numeric types with type labels is currently
+   supported by a given command, this feature will be mentioned on
+   that command's doc page. Or, for input script commands, type labels
+   can be processed using :doc:`variable <variable>` syntax using
+   labelmap functions.
+
+.. note::
+
+   Type labels cannot start with a number.
+
+The label map of only one type of interaction (atom, bond, etc.) may
+be modified by a given *labelmap* command. Any number of
+numeric-type/type-label pairs may follow. If a type label already
+exists for a given numeric type, it will be overwritten. Assigning the
+same type label to multiple numeric types is permitted; how this
```

Comment:
@jrgissing Is this correct? `Assigning the same type label to multiple numeric types is permitted;`
I thought it should be the other way, which is what other codes use. We have C1, C2, C3, ... which are all the same C type.

Reply:
@jrgissing if you assign the same label to multiple numeric types, how do you know which one is which in the current implementation?

Reply:
assigning the same label to multiple types was permitted as part of our effort to keep type labels as general as possible. currently, the first instance of a label is returned (i.e. corresponding to the lowest numeric type). this behavior should be stated in the docs, for example when using 'labelmap functions' of the variable command.

Reply:
update: type labels now required to be unique